### PR TITLE
Allow overriding `resolution` and `workspace` fields in pubspec_overrides.

### LINKED ISSUE #4393: `dart pub unpack` fails with dependency published from a workspace
# Environment
```
Dart SDK version: 3.5.3 (stable) (Wed Sep 11 16:22:47 2024 +0000) on "linux_x64"
```

# Problem
 
`unpack` command runs `dart pub get` in dependency that is a workspace and fails.

## Expected behavior

`unpack` should work "out of the box" with `resolution: workspace` packages

`dart pub unpack binary_serializable` should succeed.

## Actual behavior
```
$ dart pub unpack binary_serializable
Downloading binary_serializable 0.3.0 to `./binary_serializable-0.3.0`... 
To explore type: cd ./binary_serializable-0.3.0
Found a pubspec.yaml at /home/elijah/Development/binary_serializable-0.3.0. But it has resolution `workspace`.
But found no workspace root including it in parent directories.

See https://dart.dev/go/pub-workspaces for more information.
```

The exit code is `66`

I am able to resolve by removing the `resolution: workspace` entry in the `pubspec.yaml` and running `dart pub get` by hand.

### PR DESCRIPTION
**Pull request by sigurdm**
And use this in  `dart pub unpack` for allowing to unpack workspace packages.

Fixes #4393

### CODE REVIEW

File: lib/src/entrypoint.dart [hunk 1]
```diff
@@ -111,15 +111,16 @@ class Entrypoint {
...
-          }) =>
-              pubspecsMet[p.canonicalize(path)] ??
-              Pubspec.load(
-                path,
-                cache.sources,
-                expectedName: expectedName,
-                allowOverridesFile: withPubspecOverrides,
-                containingDescription: RootDescription(path),
-              ),
+          }) {
```
Comment (jonasfj) [inline lib/src/entrypoint.dart:114]: is this just reformatting?

File: test/unpack_test.dart [hunk 2]
```diff
@@ -125,4 +125,41 @@ Resolving dependencies in `../foo-1.2.3-pre`...
...
+    await runPub(
+      args: ['unpack', 'foo:1.0.0'],
+      output: allOf(
+        contains('Downloading foo 1.0.0 to `.${s}foo-1.0.0`...'),
+        contains(
+          '+ bar',
+        ),
+      ),
+      environment: {'_PUB_TEST_SDK_VERSION': '3.5.0'},
+    );
```
Comment (jonasfj) [inline test/unpack_test.dart:163]: await file('appPath/pubspec_overrides.yaml', 'resolution:').exists()

Or something like that, not necessary, just a suggestion.

**Review by jonasfj — APPROVED** [review]

File: lib/src/entrypoint.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/entrypoint.dart:114]: Yeah - I had some debug prints in here I think. I'll revert this part.

File: test/unpack_test.dart (hunk 2 above)
Comment (sigurdm) [reply test/unpack_test.dart:163]: Done

**Pull request merged**